### PR TITLE
Update to use Swift 4.2 and Xcode 10.2

### PR DESCRIPTION
**Pull request by RuiAAPeres**
I will then cut a release `0.6.0`.

### CODE REVIEW

**Review by andersio — APPROVED** [review]

**Review by andersio — DISMISSED** [review]
Please also update SwiftPM.

ref: #40

**Comment by RuiAAPeres** [comment]
Didn't notice we had another PR for this. Pointless to open this.

**Comment by RuiAAPeres** [comment]
@andersio fixed.

**Review by dmcrodrigues — APPROVED** [review]

File: Podfile [hunk 1]
```diff
@@ -3,7 +3,13 @@ install! 'cocoapods',
...
 
 def shared_pods
-    pod "ReactiveSwift", "~> 4.0"
+    pod "ReactiveSwift", "~> 5.0"
+end
+
+target "Example" do
+  platform :ios, "8.0"
+  pod "Kingfisher"
+  shared_pods
```
Comment (RuiAAPeres) [inline Podfile:12]: there are some oddities here. For instance, RAC was removed from the target `Example` and everything works fine. Now the thing here is that we are linking RAC from carthage, while at the same time using CocoaPods. This will have to be fixed at a later point.

File: ReactiveFeedback.xcodeproj/xcshareddata/xcschemes/Example.xcscheme [hunk 2]
```diff
@@ -80,7 +80,7 @@
             buildForTesting = "YES"
             buildForRunning = "YES"
             buildForProfiling = "YES"
-            buildForArchiving = "YES"
+            buildForArchiving = "NO"
```
Comment (diegopetrucci) [inline ReactiveFeedback.xcodeproj/xcshareddata/xcschemes/Example.xcscheme:83]: Was this change intentional?

Comment (RuiAAPeres) [reply ReactiveFeedback.xcodeproj/xcshareddata/xcschemes/Example.xcscheme:83]: Yes, otherwise this: https://github.com/SwipeCellKit/SwipeCellKit/issues/297

**Review by diegopetrucci — APPROVED** [review]

**Review by gtsiap — APPROVED** [review]

File: ReactiveFeedback.xcodeproj/xcshareddata/xcschemes/Example.xcscheme (hunk 2 above)
Comment (RuiAAPeres) [reply ReactiveFeedback.xcodeproj/xcshareddata/xcschemes/Example.xcscheme:83]: This is for the Example target, so it wouldn't impact RAF consumers.

**Review by inamiy — APPROVED** [review]

File: Podfile [hunk 3]
```diff
@@ -14,12 +20,13 @@ end
...
-target "Example" do
-    platform :ios, "10.0"
-    shared_pods
-    pod "Kingfisher", "~> 4.0"
-    pod "ReactiveCocoa", "~> 8.0"
-end
+post_install do |installer|
+  installer.pods_project.targets.each do |target|
+    target.build_configurations.each do |config|
+      config.build_settings['SWIFT_VERSION'] = '4.2'
```
Comment (mluisbrown) [inline Podfile:29]: Is this because dependent pods don't have the swift version set correctly?

**Review by mluisbrown — APPROVED** [review]
LGTM 👍

**Comment by RuiAAPeres** [comment]
@andersio I am going to dismiss your review, since what you request has been included.

File: Podfile (hunk 3 above)
Comment (RuiAAPeres) [reply Podfile:29]: This need to be reviewed at a later PR.
